datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -628,6 +627,103 @@ fn infer_join_predicates(
         .collect::<Result<Vec<_>>>()
 }
 
+/// Check whether the given expression depends only on the columns in the given schema.
+/// Meaning for a certain
```

Comment:
this sentence appears to end abruptly

Reply:
Also this appears to be a copy of https://github.com/apache/datafusion/blob/b098893a34f83f1a1df290168377d7622938b3f5/datafusion/core/src/datasource/listing/helpers.rs#L55-L54

Perhaps instead of copy/pasting the code, you could move it to a common location (e.g. datafusion-expr)?

Reply:
Hey,
I agree that this code is a duplicate and it should be a common function, however `expr_applicable_for_cols` uses column names as argument, so in case we have two columns with the same name but with different qualifiers, the function will produce incorrect result.
IMO we would need to dig into `expr_applicable_for_cols`'s usage (there's only one) and understand if it should be fixed. Unfortunately I won't have much time to do it in the coming weeks.